[improve][doc] Add a limitation for key_shared subscription type (#15709)
apache/pulsar: site2/docs/concepts-messaging.md

</Tabs>
````

:::caution
**Limitations of Key_Shared type** 

When you use Key_Shared type, be aware that:
  * You need to specify a key or orderingKey for messages.
  * You cannot use cumulative acknowledgment with Key_Shared type.

>   * When the position of the newest message in a topic is `X`, a key-shared consumer that is newly attached to the same subscription and connected to the topic will **not** receive any messages until the position of the oldest unacknowledged message in this subscription is newer than or equal to `X`. 

Reviewer: ```suggestion
  * When the position of the newest message in a topic is `X`, a key-shared consumer that is newly attached to the same subscription and connected to the topic will **not** receive any messages until all the messages before `X` have been acknowledged. 
```

Message deduplication makes Pulsar an ideal messaging system to be used in conjunction with stream processing engines (SPEs) and other systems seeking to provide effectively-once processing semantics. Messaging systems that do not offer automatic message deduplication require the SPE or other system to guarantee deduplication, which means that strict message ordering comes at the cost of burdening the application with the responsibility of deduplication. With Pulsar, strict ordering guarantees come at no application-level cost.

:::tip

> You can find more in-depth information in [this post](https://www.splunk.com/en_us/blog/it/exactly-once-is-not-exactly-the-same.html).

Reviewer 2: ```suggestion

```
Suggest removing this. 
To keep neutral and not to be challenged by ASF, we avoid adding blogs of commercial companies to the Pulsar documentation.

Author: OK. Removed.
